hassio/src/snapshots/hassio-snapshot.js

```diff
@@ -282,14 +279,25 @@ class HassioSnapshot extends PolymerElement {
       );
   }
 
-  _computeDownloadUrl(snapshotSlug) {
-    const password = encodeURIComponent(this.hass.connection.options.authToken);
-    return `/api/hassio/snapshots/${snapshotSlug}/download?api_password=${password}`;
-  }
-
-  _computeDownloadName(snapshot) {
-    const name = this._computeName(snapshot).replace(/[^a-z0-9]+/gi, "_");
-    return `Hass_io_${name}.tar`;
+  async _downloadClicked() {
+    try {
+      const signedPath = await getSignedPath(
+        this.hass,
+        `/api/hassio/snapshots/${this.snapshotSlug}/download`
+      );
+      const name = this._computeName(this.snapshot).replace(
+        /[^a-z0-9]+/gi,
+        "_"
+      );
+      const a = document.createElement("A");
+      a.href = signedPath.path;
+      a.download = `Hass_io_${name}.tar`;
+      this.$.dialog.appendChild(a);
+      a.click();
+      this.$.dialog.removeChild(a);
+    } catch (err) {
```

Comment:
Final thing: I prefer the catch around the part that can raise. So around the `await`